packages/canon/src/components/Button/Button.tsx

```diff
@@ -44,15 +31,28 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       children,
     } = props;
 
+    const { getResponsiveValue } = useCanon();
+
+    // Get the responsive value for the variant
+    const responsiveSize = getResponsiveValue(size);
+    const responsiveVariant = getResponsiveValue(variant);
+
     return (
       <button
         {...props}
         ref={ref}
         disabled={disabled}
-        className={`button ${variant} ${size}`}
+        className={[
+          'cn-button',
+          `cn-button-${responsiveSize}`,
+          `cn-button-${responsiveVariant}`,
+        ].join(' ')}
       >
         <span
-          className={['button-content', iconStart && iconEnd ? 'icon-both' : '']
+          className={[
+            'cn-button-content',
+            iconStart && iconEnd ? 'cn-button-content-icon-both' : '',
+          ]
```

Comment:
Gonna need to introduce more explicit separators here tbh, or I think we're gonna end up with naming clashes quite soon. Stuff like `cn-button__content` for example

Reply:
Yes let's validate together the naming convention and I can do a separate PR for all components.